tests/test_absolute_hydration.py

```diff
@@ -0,0 +1,166 @@
+
+
+def test_smc_parameter_change_vacuum():
+    """
+    Tests correctness of using SMC to switch between parameters.
+    The SMC predicted dg ff0 -> ff1 should match results from
+    BAR using independent sampling.
+
+    """
+    seed = 2022
+    np.random.seed(seed)
+
+    # Reduce these so the test is faster
+    num_batches = 10
+    n_md_steps = 10
+    temperature = 300.0
+    kBT = BOLTZ * temperature
+
+    mol, _ = testsystems.ligands.get_biphenyl()
+    ff0 = get_ff_am1ccc()
+
+    # shift the ff parameters
+    # should be small enough so that 1-EXP is still valid
+    ff1 = get_ff_am1ccc()
+    ff1.q_handle.params += 0.2
+    box = np.eye(3) * 1000.0
+
+    # generate vacuum samples for each ff
+    def generate_samples(mol, ff) -> Tuple[enhanced.VacuumState, smc.Samples]:
+        state = enhanced.VacuumState(mol, ff)
+        _vacuum_xv_samples, vacuum_log_weights = enhanced.generate_log_weighted_samples(
+            mol,
+            temperature,
+            state.U_easy,
+            state.U_full,
+            seed,
+            num_batches=num_batches,
+            num_workers=1,
+            steps_per_batch=n_md_steps,
+        )
+
+        # discard velocities: (x, v) -> x
+        vacuum_samples = _vacuum_xv_samples[:, 0, :]
+        vacuum_vels = _vacuum_xv_samples[:, 1, :]
+        idxs = smc.multinomial_resample(vacuum_log_weights)[0]
+        vacuum_samples = [CoordsVelBox(coords=vacuum_samples[i], velocities=vacuum_vels[i], box=box) for i in idxs]
+        return state, vacuum_samples
+
+    state0, vacuum_samples0 = generate_samples(mol, ff0)
+    state1, vacuum_samples1 = generate_samples(mol, ff1)
+
+    # set up a system for sampling the new parameters using SMC
+    samples, lambdas, propagate, log_prob, resample = absolute_hydration.set_up_ahfe_system_for_smc_parameter_changes(
+        mol,
+        n_walkers=num_batches,
+        n_md_steps=n_md_steps,
+        resample_thresh=0.6,
+        initial_samples=vacuum_samples0,
+        ff0=ff0,
+        ff1=ff1,
+        is_vacuum=True,
+        n_windows=10,
+    )
+
+    smc_result = smc.sequential_monte_carlo(samples, lambdas, propagate, log_prob, resample)
+    smc_dg = one_sided_exp(-smc_result["log_weights_traj"][-1])
+
+    # BAR from independent runs
+    def batched_u_fxn(samples, state):
+        # Only compute vacuum energies
+        return np.array([state.U_full(x.coords) for x in samples]) / kBT
+
+    u_fwd = batched_u_fxn(vacuum_samples0, state1) - batched_u_fxn(vacuum_samples0, state0)
+    u_rev = batched_u_fxn(vacuum_samples1, state0) - batched_u_fxn(vacuum_samples1, state1)
+    bar_dg = pymbar.BAR(u_fwd, u_rev, compute_uncertainty=False)
+    assert smc_dg == pytest.approx(bar_dg, abs=1e-1)
+
+
+def test_smc_parameter_change_solvent():
+    """
+    Tests correctness of using SMC to switch between parameters.
+    The SMC predicted dg ff0 -> ff1 should match results from
+    BAR using independent sampling.
```

Comment:
Another possible test is to predict in both directions `ff0 -> ff1` and `ff1 -> ff0`